library/tiqr/Tiqr/StateStorage/Pdo.php

```diff
@@ -68,12 +68,13 @@ private function keyExists($key)
         if ($sth->execute(array($key))) {
             return $sth->fetchColumn();
         }
-        $this->logger->error('The state storage key could not be found in the database');
+        $this->logger->info('The state storage key could not be found in the database');
     }
 
     private function cleanExpired() {
         $sth = $this->handle->prepare("DELETE FROM ".$this->tablename." WHERE `expire` < ? AND NOT `expire` = 0");
-        if (!$sth->execute(array(time()))){
+        $result = $sth->execute(array(time()));
+        if (!$result || $sth->rowCount() === 0){
             $this->logger->error('Unable to remove expired keys from the pdo state storage');
```

Comment:
Is there a reason for the normal log instead of an exception?

Reply:
As we dicussed IRL (but stated here for the record)

Given this method is not disturbing the process (it is called to clean up expired records). It is not a problem when it fails. The next call should perform the same task again. No data is lost, the application can continue without issue.

I will add a comment to explain this in code